@here/harp-datasource-protocol/lib/Theme.ts

```diff
@@ -183,6 +184,157 @@ export interface BaseStyle {
     labelProperty?: string;
 }
 
+/**
+ *
+ * @defaultSnippets [
+ *     {
+ *         "label": "New solid-line",
+ *         "description": "Add a New 'solid-line' Styling Rule",
+ *         "body": {
+ *             "technique": "solid-line",
+ *             "when": "$1",
+ *             "attr": {
+ *                 "color": "#$2fff",
+ *                 "lineWidth": "^${3:1}"
+ *             }
+ *         }
+ *     },
+ *     {
+ *         "label": "New dashed-line",
+ *         "description": "Add a New 'dashed-line' Styling Rule",
+ *         "body": {
+ *             "technique": "dashed-line",
+ *             "when": "$1",
+ *             "attr": {
+ *                 "color": "#$2fff",
+ *                 "lineWidth": "^${3:1}"
```

Comment:
Aren't you supposed to add the ``dashSize`` and ``gapSize`` properties here too?

Reply:
Yep, think I should insert some special properties for every technique.